### DISCUSSION TITLE
Non-canonical SMARTS pattern in _BaseIonic class

### DISCUSSION THREAD

**Discussion by radifar** [chemosim-lab/ProLIF#106] [Q&A]
Hi Cedric, I'm trying to find some alternative in determining the group charge in my PyPLIF HIPPOS project. And I'm interested in using SMARTS pattern like in ProLIF and Arpeggio. Unfortunately as I'm using Open Babel, it can not read the SMARTS pattern from ProLIF and gave the following error:

![prolif_smarts_error](https://user-images.githubusercontent.com/11450417/217376021-3803b05f-ed50-460c-81fe-ae37e7aac5a3.png)

I try to figure out why does Open Babel can not read the SMARTS pattern from ProLIF? Then I remember that ProLIF uses RDKit, and perhaps it uses some non-canonical SMARTS pattern. Now I am wondering how can I convert this pattern into canonical Daylight SMARTS so that I can use this in Open Babel.

Thanks,
Radif

**Comment by cbouy** [comment]
Hi @radifar,

Indeed, there are "extensions" for SMARTS that add some features to the Daylight one in RDKit.
Here, the `{1-}` after the charge is a [range query](https://www.rdkit.org/docs/RDKit_Book.html#range-queries) which translates as "at least 1" so `[+{1-}]` could be replaced by `[+1,+2,+3]` and so on. Unfortunately [OpenBabel does not seem to support this specific extension](https://openbabel.org/wiki/SMARTS).

Hope this helps,
Cedric

**Reply by radifar** [reply]
Thank you for the pointer, that helps a lot!
